py/builtinimport.c

```diff
@@ -145,11 +145,16 @@ STATIC void do_load_from_lexer(mp_obj_t module_obj, mp_lexer_t *lex) {
 #endif
 
 #if MICROPY_PERSISTENT_CODE_LOAD || MICROPY_MODULE_FROZEN_MPY
-STATIC void do_execute_raw_code(mp_obj_t module_obj, mp_raw_code_t *raw_code) {
+STATIC void do_execute_raw_code(mp_obj_t module_obj, mp_raw_code_t *raw_code
     #if MICROPY_PY___FILE__
-    // TODO
-    //qstr source_name = lex->source_name;
-    //mp_store_attr(module_obj, MP_QSTR___file__, MP_OBJ_NEW_QSTR(source_name));
+    , const char* source_name
+    #endif
+) {
+    #if MICROPY_PY___FILE__
+    if (source_name != NULL) {
```

Comment:
Can source_name ever be NULL?

Reply:
dunno, maybe if accessed from a C module ? i crash so much my C code that i'd better be safe than sorry. I leave you sole judge :p

Reply:
It needs to be understood if it can be NULL (or empty) or not.  I guess it must be non-NULL and non-empty.

Reply:
if .__file__  attr exists but is None then it can't be printed. But that's probably not possible there